main.go

```diff
@@ -327,10 +331,128 @@ func (c *CASProxy) ViceSubdomain(r *http.Request, m *mux.RouteMatch) bool {
+}
+
+const lookupExternalUUIDQuery = `
+query ExternalID($subdomain: String) {
+	jobs(where: {subdomain: {_eq: $subdomain}}) {
+		steps: jobStepssByjobId {
+			external_id
+		}
+	}
+}
+`
+
+// lookupExternalUUID returns the external job UUID associated with the
+// subdomain.
+func (c *CASProxy) lookupExternalUUID(subdomain string) (string, error) {
+	var (
+		err error
+		ok  bool
+	)
+
+	client := graphql.NewClient(c.graphqlBase)
+	req := graphql.NewRequest(lookupExternalUUIDQuery)
+	req.Var("subdomain", subdomain)
+
+	data := map[string][]map[string][]map[string]string{}
+
+	if err = client.Run(context.Background(), req, &data); err != nil {
+		return "", nil
+	}
+
+	if _, ok = data["jobs"][0]["steps"][0]["external_id"]; !ok {
+		return "", fmt.Errorf("missing external_id for job with subdomain '%s'; data => '%+v'", subdomain, data)
+	}
+	return data["jobs"][0]["steps"][0]["external_id"], nil
+}
+
+const lookupJobStatusUpdatesQuery = `
+query StatusUpdates($external_id: String) {
+  job_status_updates(
+    where: {
+      external_id: {_eq: $external_id}
+    },
+    order_by: sent_on_desc
+  ) {
+    id
+    status
+    message
+    sent_on
+  }
+}
+`
+
+// JobStatusUpdate contains the fields we need/want from a job status update.
+type JobStatusUpdate struct {
+	Status  string `json:"status"`
+	SentOn  int64  `json:"sent_on"`
+	UUID    string `json:"id"`
+	Message string `json:"message"`
+}
+
+// LookupJobStatusUpdates returns the list of job status updates in reverse
+// chronological order.
+func (c *CASProxy) LookupJobStatusUpdates(w http.ResponseWriter, r *http.Request) {
+	u := r.FormValue("url")
+
+	subdomain, err := extractSubdomain(u)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error getting subdomain for URL %s", u), http.StatusInternalServerError)
+		return
+	}
+	if subdomain == "" {
+		http.Error(w, fmt.Sprintf("empty subdomain for URL '%s'", u), http.StatusBadRequest)
+		return
+	}
+
+	externalID, err := c.lookupExternalUUID(subdomain)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error getting external UUID for subdomain %s", subdomain), http.StatusInternalServerError)
+		return
+	}
+	if externalID == "" {
+		http.Error(w, fmt.Sprintf("empty external UUID for subdomain %s", subdomain), http.StatusInternalServerError)
+		return
+	}
+
+	client := graphql.NewClient(c.graphqlBase)
+	req := graphql.NewRequest(lookupJobStatusUpdatesQuery)
+	req.Var("external_id", externalID)
+
+	data := map[string][]JobStatusUpdate{}
+
+	if err = client.Run(context.Background(), req, &data); err != nil {
+		http.Error(w, fmt.Sprintf("error from job status updates query for subdomain %s: %s", subdomain, err), http.StatusInternalServerError)
+		return
+	}
+
+	js, err := json.Marshal(data)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error marshalling response from job status updates query for subdomain %s: %s", subdomain, err), http.StatusInternalServerError)
+		return
+	}
+
+	fmt.Fprintln(w, string(js))
```

Comment:
Might the JSON be used for something other than visual inspection?

Reply:
`fmt.Fprintln()` is being used here to write the JSON response body over the wire using the `http.ResponseWriter` passed into the function. It's not actually logging the response, though the code kinda looks like it is.